tls/s2n_tls13_secrets.c

```diff
@@ -671,3 +694,53 @@ S2N_RESULT s2n_tls13_secrets_get(struct s2n_connection *conn, s2n_extract_secret
     RESULT_ENSURE_GT(secret->size, 0);
     return S2N_RESULT_OK;
 }
+
+int s2n_connection_tls_exporter(
+        struct s2n_connection *conn,
+        uint8_t *output_,
+        uint32_t output_length,
+        const uint8_t *label_,
+        uint32_t label_length,
+        const uint8_t *context,
+        uint32_t context_length)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(output_);
+    POSIX_ENSURE_REF(label_);
+    POSIX_ENSURE_REF(context);
+    POSIX_ENSURE(s2n_connection_get_protocol_version(conn) == S2N_TLS13, S2N_ERR_INVALID_STATE);
+
+    s2n_hmac_algorithm hmac_alg = conn->secure->cipher_suite->prf_alg;
+
+    uint8_t size = 0;
+    if (s2n_hmac_digest_size(hmac_alg, &size) != S2N_SUCCESS) {
+        size = 0;
+    }
+    POSIX_ENSURE_GT(size, 0);
+    struct s2n_blob exporter_master_secret = (struct s2n_blob){ .data = conn->secrets.version.tls13.exporter_master_secret, .size = size };
+    const struct s2n_blob label = (struct s2n_blob){ .data = (uint8_t *) label_, .size = label_length };
+
+    uint8_t derived_secret_bytes[S2N_TLS13_SECRET_MAX_LEN] = { 0 };
+    struct s2n_blob derived_secret = { 0 };
+    POSIX_GUARD(s2n_blob_init(&derived_secret, derived_secret_bytes, S2N_TLS13_SECRET_MAX_LEN));
+    POSIX_GUARD_RESULT(s2n_derive_secret(hmac_alg, &exporter_master_secret, &label, &CONN_HASH(conn, transcript_hash_digest), &derived_secret));
+
+    DEFER_CLEANUP(struct s2n_hmac_state hmac_state = { 0 }, s2n_hmac_free);
+    POSIX_GUARD(s2n_hmac_new(&hmac_state));
+
+    DEFER_CLEANUP(struct s2n_hash_state hash = { 0 }, s2n_hash_free);
+    POSIX_GUARD(s2n_hash_new(&hash));
+
+    s2n_hash_algorithm hash_alg;
```

Comment:
I recognize that this is unnecessary, but we're paranoid and don't like thinking about whether or not things actually need to be initialized in case we get it wrong. We initialize everything just in case. We build in some environments that actually don't initialize for you, and it's bitten us.
```suggestion

    s2n_hash_algorithm hash_alg = { 0 };
```